src/fields/abstractField.js

```diff
@@ -72,6 +84,22 @@ export default {
 	},
 
 	methods: {
+		deleteDataOnHide() {
+			// emits hidden event up with field, if should be deleted, if should set initial back, or if should keep on hide.
+			let field = this.schema;
+			let visible = isFieldVisible(this.model, field, this);
+
+			if (!this.formOptions.deleteDataOnHide) {
+				return false;
+			}
+
+			this.$emit("hidden", {
+				field: field,
+				shouldDelete: this.formOptions.deleteDataOnHide && !visible,
+				shouldSetInitial: this.formOptions.deleteDataOnHide && visible && has(field, "initial"),
+				keepOnHide: field.keepOnHide
+			});
```

Comment:
I am wondering if it makes sense to unconditionally emit this event, and simply have `field` `visible` and `keepOnHide` as the keys. And remove all notions of `formOptions.deleteDataOnHide` since all of that will be handled by SurefyreForm. Let's discuss.